buildSrc/src/main/kotlin/com.hedera.block.protobuf.gradle.kts

```
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.protobuf.gradle.id
+import com.hedera.block.tasks.GitClone
+
+plugins {
+    id("java-library")
+    id("com.hedera.block.conventions")
+    id("com.google.protobuf")
+}
+
+tasks.register<GitClone>("cloneHederaProtobufs") {
+    url = "https://github.com/hashgraph/hedera-protobufs.git"
+    offline = gradle.startParameter.isOffline
+    localCloneDirectory = layout.buildDirectory.dir("hedera-protobufs")
+}
+
+// Configure Protobuf Plugin to download protoc executable rather than using local installed version
+protobuf {
+    val libs = the<VersionCatalogsExtension>().named("libs")
+    protoc { artifact = "com.google.protobuf:protoc:" + libs.findVersion("google.proto").get() }
+    plugins {
+        // Add GRPC plugin as we need to generate GRPC services
+        id("grpc") {
+            artifact =
+                "io.grpc:protoc-gen-grpc-java:" + libs.findVersion("grpc.protobuf.grpc").get()
+        }
+    }
+    generateProtoTasks { ofSourceSet("main").forEach { it.plugins { id("grpc") } } }
```

```suggestion
    generateProtoTasks { all().forEach { it.plugins { id("grpc") } } }
```

Use `all` here and then make the decision what is used for which Source Set in `stream/build.gradle.kts` (see comment there).

Will do a similar change in services repo at some point (https://github.com/hashgraph/hedera-services/pull/14026).

---

Fixed